File: app_java_wiki_chat/spring-backend/src/main/java/com/platform/securelearning/session/Session.java
package com.platform.securelearning.session;

import java.time.Instant;

/**
 * An anonymous chat participant (SR-2). Deliberately carries nothing but an opaque,
 * randomly generated id and a room assignment — no username, email, or IP address is
 * ever attached to a Session, so even a compromised backend has no PII to leak.
 */
public record Session(String sessionId, String roomCode, Instant joinedAt) {
}

File: app02_csharp_wiki_chat/aspnet-backend/SecureLearning.Backend/Sessions/Session.cs
namespace SecureLearning.Backend.Sessions;

/// <summary>
/// An anonymous chat participant (SR-2). Carries nothing but an opaque, randomly
/// generated id and a room assignment — no username, email, or IP address is ever
/// attached, so even a compromised backend has no PII to leak.
/// </summary>
public sealed record Session(string SessionId, string RoomCode, DateTimeOffset JoinedAt);
